src/forms/renderers/MultiLineSecret.tsx

```diff
@@ -0,0 +1,139 @@
+    const { handleChange, path, uischema } = props;
+
+    const [isUploadOpen, setIsUploadOpen] = useState(false);
+
+    const submitFile = useCallback(
+        (acceptedFiles) => {
+            if (acceptedFiles) {
+                const uploadDone = (inputValue: string | null) => {
+                    handleChange(path, inputValue);
+                    setIsUploadOpen(false);
+                };
+
+                acceptedFiles.forEach(async (file: File) => {
+                    const reader = new FileReader();
+
+                    reader.onabort = () => uploadDone(null);
+                    reader.onerror = () => uploadDone(null);
+                    reader.onloadend = () => {
+                        const result = reader.result ?? null;
+                        uploadDone(result as string);
+                    };
+                    reader.readAsText(file);
+                });
+            }
+
+            setIsUploadOpen(false);
+        },
+        [handleChange, path]
+    );
+
+    // Make it a password again so the input is masked
+    uischema.options.format = 'password';
+
+    return (
+        <>
+            <Stack direction="row" spacing={2} sx={{ alignItems: 'center' }}>
+                <MaterialInputControl
+                    {...props}
+                    muiInputProps={{
+                        onPaste: (event: any) => {
+                            // Stop the event so the change does no fire and strip out the newlines
+                            event.stopPropagation();
+                            event.preventDefault();
+
+                            // Go ahead and update the value
+                            handleChange(
+                                path,
+                                event.clipboardData.getData('text/plain')
+                            );
+                        },
+                    }}
+                    input={MuiInputText}
+                />
+
+                <Box>
+                    <Button
+                        sx={{ whiteSpace: 'nowrap' }}
+                        onClick={() => setIsUploadOpen(!isUploadOpen)}
+                    >
+                        Use file (.pem)
+                    </Button>
+                </Box>
+            </Stack>
+
+            <DropzoneDialog
+                open={isUploadOpen}
+                onSave={submitFile}
+                filesLimit={1}
+                // List taken from -> https://pki-tutorial.readthedocs.io/en/latest/mime.html
+                acceptedFiles={[
+                    '.pem',
+                    '.p8',
+                    '.p10',
+                    '.p12',
+                    '.key',
+                    '.csr',
+                    '.cer',
+                    '.cr1',
+                    '.crt',
+                    '.p7c',
+                    '.p7b',
+                    '.p7r',
+                    '.der',
+                    '.pfx',
+                    '.spc',
+                ]}
```

Comment:
@psFried - I'd like to chat with you above this before we launch MVP. I'm not totally sure what list of files we should  allow, should this be in the UI or the schema, should we add an option to the schema for "fileUpload", etc.

Reply:
@travjenkins AFAIK private ssh keys generated using `ssh-keygen` do not have an extension, they are just called something like `id_rsa`, so I'm not sure, but would this force the user to rename their private key to add an extension?

Reply:
That's a good point. It would. So I think I will add no file type to the list as well.